packages/transaction-controller/src/utils/simulation.ts

```diff
@@ -284,41 +289,46 @@ async function getTokenBalanceChanges(
   request: GetSimulationDataRequest,
   events: ParsedEvent[],
 ): Promise<SimulationTokenBalanceChange[]> {
-  const balanceTransactionsByToken = getTokenBalanceTransactions(
-    request,
-    events,
-  );
+  const balanceTxs = getTokenBalanceTransactions(request, events);
 
-  const balanceTransactions = [...balanceTransactionsByToken.values()];
+  const balanceTransactions = [
```

Comment:
This is no longer the balance transactions since it includes the request itself, so should we just call this `transactions`?